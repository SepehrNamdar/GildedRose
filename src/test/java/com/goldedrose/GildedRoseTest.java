package com.goldedrose;

import org.junit.jupiter.api.Test;

import static org.approvaltests.combinations.CombinationApprovals.verifyAllCombinations;

class GildedRoseTest {

    @Test
    void should_update_quality() {
        String[] name =
                new String[] {null, "", "foo", "Aged Brie", "Backstage passes to a TAFKAL80ETC concert", "Sulfuras, Hand of Ragnaros"};
        Integer[] sellIn = new Integer[] {null, 0, -1, 10, 11, 12, 13, 14};
        Integer[] quality = new Integer[] {null, 0, 1, 50, 49, 51};

        verifyAllCombinations(this::verifyUpdateQuality, name, sellIn, quality);
    }

    private String verifyUpdateQuality(String name, int sellIn, int quality) {
        Item[] items = new Item[] { new Item(name, sellIn, quality) };
        GildedRose app = new GildedRose(items);

        app.updateQuality();

        return app.items[0].toString();
    }
}
