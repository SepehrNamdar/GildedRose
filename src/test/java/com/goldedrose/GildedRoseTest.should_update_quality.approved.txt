[null, null, null] => java.lang.NullPointerException: null 
[null, null, 0] => java.lang.NullPointerException: null 
[null, null, 1] => java.lang.NullPointerException: null 
[null, null, 50] => java.lang.NullPointerException: null 
[null, null, 49] => java.lang.NullPointerException: null 
[null, null, 51] => java.lang.NullPointerException: null 
[null, 0, null] => java.lang.NullPointerException: null 
[null, 0, 0] => java.lang.NullPointerException: null 
[null, 0, 1] => java.lang.NullPointerException: null 
[null, 0, 50] => java.lang.NullPointerException: null 
[null, 0, 49] => java.lang.NullPointerException: null 
[null, 0, 51] => java.lang.NullPointerException: null 
[null, -1, null] => java.lang.NullPointerException: null 
[null, -1, 0] => java.lang.NullPointerException: null 
[null, -1, 1] => java.lang.NullPointerException: null 
[null, -1, 50] => java.lang.NullPointerException: null 
[null, -1, 49] => java.lang.NullPointerException: null 
[null, -1, 51] => java.lang.NullPointerException: null 
[null, 10, null] => java.lang.NullPointerException: null 
[null, 10, 0] => java.lang.NullPointerException: null 
[null, 10, 1] => java.lang.NullPointerException: null 
[null, 10, 50] => java.lang.NullPointerException: null 
[null, 10, 49] => java.lang.NullPointerException: null 
[null, 10, 51] => java.lang.NullPointerException: null 
[null, 11, null] => java.lang.NullPointerException: null 
[null, 11, 0] => java.lang.NullPointerException: null 
[null, 11, 1] => java.lang.NullPointerException: null 
[null, 11, 50] => java.lang.NullPointerException: null 
[null, 11, 49] => java.lang.NullPointerException: null 
[null, 11, 51] => java.lang.NullPointerException: null 
[null, 12, null] => java.lang.NullPointerException: null 
[null, 12, 0] => java.lang.NullPointerException: null 
[null, 12, 1] => java.lang.NullPointerException: null 
[null, 12, 50] => java.lang.NullPointerException: null 
[null, 12, 49] => java.lang.NullPointerException: null 
[null, 12, 51] => java.lang.NullPointerException: null 
[null, 13, null] => java.lang.NullPointerException: null 
[null, 13, 0] => java.lang.NullPointerException: null 
[null, 13, 1] => java.lang.NullPointerException: null 
[null, 13, 50] => java.lang.NullPointerException: null 
[null, 13, 49] => java.lang.NullPointerException: null 
[null, 13, 51] => java.lang.NullPointerException: null 
[null, 14, null] => java.lang.NullPointerException: null 
[null, 14, 0] => java.lang.NullPointerException: null 
[null, 14, 1] => java.lang.NullPointerException: null 
[null, 14, 50] => java.lang.NullPointerException: null 
[null, 14, 49] => java.lang.NullPointerException: null 
[null, 14, 51] => java.lang.NullPointerException: null 
[, null, null] => java.lang.NullPointerException: null 
[, null, 0] => java.lang.NullPointerException: null 
[, null, 1] => java.lang.NullPointerException: null 
[, null, 50] => java.lang.NullPointerException: null 
[, null, 49] => java.lang.NullPointerException: null 
[, null, 51] => java.lang.NullPointerException: null 
[, 0, null] => java.lang.NullPointerException: null 
[, 0, 0] => , -1, 0 
[, 0, 1] => , -1, 0 
[, 0, 50] => , -1, 48 
[, 0, 49] => , -1, 47 
[, 0, 51] => , -1, 49 
[, -1, null] => java.lang.NullPointerException: null 
[, -1, 0] => , -2, 0 
[, -1, 1] => , -2, 0 
[, -1, 50] => , -2, 48 
[, -1, 49] => , -2, 47 
[, -1, 51] => , -2, 49 
[, 10, null] => java.lang.NullPointerException: null 
[, 10, 0] => , 9, 0 
[, 10, 1] => , 9, 0 
[, 10, 50] => , 9, 49 
[, 10, 49] => , 9, 48 
[, 10, 51] => , 9, 50 
[, 11, null] => java.lang.NullPointerException: null 
[, 11, 0] => , 10, 0 
[, 11, 1] => , 10, 0 
[, 11, 50] => , 10, 49 
[, 11, 49] => , 10, 48 
[, 11, 51] => , 10, 50 
[, 12, null] => java.lang.NullPointerException: null 
[, 12, 0] => , 11, 0 
[, 12, 1] => , 11, 0 
[, 12, 50] => , 11, 49 
[, 12, 49] => , 11, 48 
[, 12, 51] => , 11, 50 
[, 13, null] => java.lang.NullPointerException: null 
[, 13, 0] => , 12, 0 
[, 13, 1] => , 12, 0 
[, 13, 50] => , 12, 49 
[, 13, 49] => , 12, 48 
[, 13, 51] => , 12, 50 
[, 14, null] => java.lang.NullPointerException: null 
[, 14, 0] => , 13, 0 
[, 14, 1] => , 13, 0 
[, 14, 50] => , 13, 49 
[, 14, 49] => , 13, 48 
[, 14, 51] => , 13, 50 
[foo, null, null] => java.lang.NullPointerException: null 
[foo, null, 0] => java.lang.NullPointerException: null 
[foo, null, 1] => java.lang.NullPointerException: null 
[foo, null, 50] => java.lang.NullPointerException: null 
[foo, null, 49] => java.lang.NullPointerException: null 
[foo, null, 51] => java.lang.NullPointerException: null 
[foo, 0, null] => java.lang.NullPointerException: null 
[foo, 0, 0] => foo, -1, 0 
[foo, 0, 1] => foo, -1, 0 
[foo, 0, 50] => foo, -1, 48 
[foo, 0, 49] => foo, -1, 47 
[foo, 0, 51] => foo, -1, 49 
[foo, -1, null] => java.lang.NullPointerException: null 
[foo, -1, 0] => foo, -2, 0 
[foo, -1, 1] => foo, -2, 0 
[foo, -1, 50] => foo, -2, 48 
[foo, -1, 49] => foo, -2, 47 
[foo, -1, 51] => foo, -2, 49 
[foo, 10, null] => java.lang.NullPointerException: null 
[foo, 10, 0] => foo, 9, 0 
[foo, 10, 1] => foo, 9, 0 
[foo, 10, 50] => foo, 9, 49 
[foo, 10, 49] => foo, 9, 48 
[foo, 10, 51] => foo, 9, 50 
[foo, 11, null] => java.lang.NullPointerException: null 
[foo, 11, 0] => foo, 10, 0 
[foo, 11, 1] => foo, 10, 0 
[foo, 11, 50] => foo, 10, 49 
[foo, 11, 49] => foo, 10, 48 
[foo, 11, 51] => foo, 10, 50 
[foo, 12, null] => java.lang.NullPointerException: null 
[foo, 12, 0] => foo, 11, 0 
[foo, 12, 1] => foo, 11, 0 
[foo, 12, 50] => foo, 11, 49 
[foo, 12, 49] => foo, 11, 48 
[foo, 12, 51] => foo, 11, 50 
[foo, 13, null] => java.lang.NullPointerException: null 
[foo, 13, 0] => foo, 12, 0 
[foo, 13, 1] => foo, 12, 0 
[foo, 13, 50] => foo, 12, 49 
[foo, 13, 49] => foo, 12, 48 
[foo, 13, 51] => foo, 12, 50 
[foo, 14, null] => java.lang.NullPointerException: null 
[foo, 14, 0] => foo, 13, 0 
[foo, 14, 1] => foo, 13, 0 
[foo, 14, 50] => foo, 13, 49 
[foo, 14, 49] => foo, 13, 48 
[foo, 14, 51] => foo, 13, 50 
[Aged Brie, null, null] => java.lang.NullPointerException: null 
[Aged Brie, null, 0] => java.lang.NullPointerException: null 
[Aged Brie, null, 1] => java.lang.NullPointerException: null 
[Aged Brie, null, 50] => java.lang.NullPointerException: null 
[Aged Brie, null, 49] => java.lang.NullPointerException: null 
[Aged Brie, null, 51] => java.lang.NullPointerException: null 
[Aged Brie, 0, null] => java.lang.NullPointerException: null 
[Aged Brie, 0, 0] => Aged Brie, -1, 2 
[Aged Brie, 0, 1] => Aged Brie, -1, 3 
[Aged Brie, 0, 50] => Aged Brie, -1, 50 
[Aged Brie, 0, 49] => Aged Brie, -1, 50 
[Aged Brie, 0, 51] => Aged Brie, -1, 51 
[Aged Brie, -1, null] => java.lang.NullPointerException: null 
[Aged Brie, -1, 0] => Aged Brie, -2, 2 
[Aged Brie, -1, 1] => Aged Brie, -2, 3 
[Aged Brie, -1, 50] => Aged Brie, -2, 50 
[Aged Brie, -1, 49] => Aged Brie, -2, 50 
[Aged Brie, -1, 51] => Aged Brie, -2, 51 
[Aged Brie, 10, null] => java.lang.NullPointerException: null 
[Aged Brie, 10, 0] => Aged Brie, 9, 1 
[Aged Brie, 10, 1] => Aged Brie, 9, 2 
[Aged Brie, 10, 50] => Aged Brie, 9, 50 
[Aged Brie, 10, 49] => Aged Brie, 9, 50 
[Aged Brie, 10, 51] => Aged Brie, 9, 51 
[Aged Brie, 11, null] => java.lang.NullPointerException: null 
[Aged Brie, 11, 0] => Aged Brie, 10, 1 
[Aged Brie, 11, 1] => Aged Brie, 10, 2 
[Aged Brie, 11, 50] => Aged Brie, 10, 50 
[Aged Brie, 11, 49] => Aged Brie, 10, 50 
[Aged Brie, 11, 51] => Aged Brie, 10, 51 
[Aged Brie, 12, null] => java.lang.NullPointerException: null 
[Aged Brie, 12, 0] => Aged Brie, 11, 1 
[Aged Brie, 12, 1] => Aged Brie, 11, 2 
[Aged Brie, 12, 50] => Aged Brie, 11, 50 
[Aged Brie, 12, 49] => Aged Brie, 11, 50 
[Aged Brie, 12, 51] => Aged Brie, 11, 51 
[Aged Brie, 13, null] => java.lang.NullPointerException: null 
[Aged Brie, 13, 0] => Aged Brie, 12, 1 
[Aged Brie, 13, 1] => Aged Brie, 12, 2 
[Aged Brie, 13, 50] => Aged Brie, 12, 50 
[Aged Brie, 13, 49] => Aged Brie, 12, 50 
[Aged Brie, 13, 51] => Aged Brie, 12, 51 
[Aged Brie, 14, null] => java.lang.NullPointerException: null 
[Aged Brie, 14, 0] => Aged Brie, 13, 1 
[Aged Brie, 14, 1] => Aged Brie, 13, 2 
[Aged Brie, 14, 50] => Aged Brie, 13, 50 
[Aged Brie, 14, 49] => Aged Brie, 13, 50 
[Aged Brie, 14, 51] => Aged Brie, 13, 51 
[Backstage passes to a TAFKAL80ETC concert, null, null] => java.lang.NullPointerException: null 
[Backstage passes to a TAFKAL80ETC concert, null, 0] => java.lang.NullPointerException: null 
[Backstage passes to a TAFKAL80ETC concert, null, 1] => java.lang.NullPointerException: null 
[Backstage passes to a TAFKAL80ETC concert, null, 50] => java.lang.NullPointerException: null 
[Backstage passes to a TAFKAL80ETC concert, null, 49] => java.lang.NullPointerException: null 
[Backstage passes to a TAFKAL80ETC concert, null, 51] => java.lang.NullPointerException: null 
[Backstage passes to a TAFKAL80ETC concert, 0, null] => java.lang.NullPointerException: null 
[Backstage passes to a TAFKAL80ETC concert, 0, 0] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 1] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 50] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 49] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 51] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, null] => java.lang.NullPointerException: null 
[Backstage passes to a TAFKAL80ETC concert, -1, 0] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 1] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 50] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 49] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 51] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, 10, null] => java.lang.NullPointerException: null 
[Backstage passes to a TAFKAL80ETC concert, 10, 0] => Backstage passes to a TAFKAL80ETC concert, 9, 2 
[Backstage passes to a TAFKAL80ETC concert, 10, 1] => Backstage passes to a TAFKAL80ETC concert, 9, 3 
[Backstage passes to a TAFKAL80ETC concert, 10, 50] => Backstage passes to a TAFKAL80ETC concert, 9, 50 
[Backstage passes to a TAFKAL80ETC concert, 10, 49] => Backstage passes to a TAFKAL80ETC concert, 9, 50 
[Backstage passes to a TAFKAL80ETC concert, 10, 51] => Backstage passes to a TAFKAL80ETC concert, 9, 51 
[Backstage passes to a TAFKAL80ETC concert, 11, null] => java.lang.NullPointerException: null 
[Backstage passes to a TAFKAL80ETC concert, 11, 0] => Backstage passes to a TAFKAL80ETC concert, 10, 1 
[Backstage passes to a TAFKAL80ETC concert, 11, 1] => Backstage passes to a TAFKAL80ETC concert, 10, 2 
[Backstage passes to a TAFKAL80ETC concert, 11, 50] => Backstage passes to a TAFKAL80ETC concert, 10, 50 
[Backstage passes to a TAFKAL80ETC concert, 11, 49] => Backstage passes to a TAFKAL80ETC concert, 10, 50 
[Backstage passes to a TAFKAL80ETC concert, 11, 51] => Backstage passes to a TAFKAL80ETC concert, 10, 51 
[Backstage passes to a TAFKAL80ETC concert, 12, null] => java.lang.NullPointerException: null 
[Backstage passes to a TAFKAL80ETC concert, 12, 0] => Backstage passes to a TAFKAL80ETC concert, 11, 1 
[Backstage passes to a TAFKAL80ETC concert, 12, 1] => Backstage passes to a TAFKAL80ETC concert, 11, 2 
[Backstage passes to a TAFKAL80ETC concert, 12, 50] => Backstage passes to a TAFKAL80ETC concert, 11, 50 
[Backstage passes to a TAFKAL80ETC concert, 12, 49] => Backstage passes to a TAFKAL80ETC concert, 11, 50 
[Backstage passes to a TAFKAL80ETC concert, 12, 51] => Backstage passes to a TAFKAL80ETC concert, 11, 51 
[Backstage passes to a TAFKAL80ETC concert, 13, null] => java.lang.NullPointerException: null 
[Backstage passes to a TAFKAL80ETC concert, 13, 0] => Backstage passes to a TAFKAL80ETC concert, 12, 1 
[Backstage passes to a TAFKAL80ETC concert, 13, 1] => Backstage passes to a TAFKAL80ETC concert, 12, 2 
[Backstage passes to a TAFKAL80ETC concert, 13, 50] => Backstage passes to a TAFKAL80ETC concert, 12, 50 
[Backstage passes to a TAFKAL80ETC concert, 13, 49] => Backstage passes to a TAFKAL80ETC concert, 12, 50 
[Backstage passes to a TAFKAL80ETC concert, 13, 51] => Backstage passes to a TAFKAL80ETC concert, 12, 51 
[Backstage passes to a TAFKAL80ETC concert, 14, null] => java.lang.NullPointerException: null 
[Backstage passes to a TAFKAL80ETC concert, 14, 0] => Backstage passes to a TAFKAL80ETC concert, 13, 1 
[Backstage passes to a TAFKAL80ETC concert, 14, 1] => Backstage passes to a TAFKAL80ETC concert, 13, 2 
[Backstage passes to a TAFKAL80ETC concert, 14, 50] => Backstage passes to a TAFKAL80ETC concert, 13, 50 
[Backstage passes to a TAFKAL80ETC concert, 14, 49] => Backstage passes to a TAFKAL80ETC concert, 13, 50 
[Backstage passes to a TAFKAL80ETC concert, 14, 51] => Backstage passes to a TAFKAL80ETC concert, 13, 51 
[Sulfuras, Hand of Ragnaros, null, null] => java.lang.NullPointerException: null 
[Sulfuras, Hand of Ragnaros, null, 0] => java.lang.NullPointerException: null 
[Sulfuras, Hand of Ragnaros, null, 1] => java.lang.NullPointerException: null 
[Sulfuras, Hand of Ragnaros, null, 50] => java.lang.NullPointerException: null 
[Sulfuras, Hand of Ragnaros, null, 49] => java.lang.NullPointerException: null 
[Sulfuras, Hand of Ragnaros, null, 51] => java.lang.NullPointerException: null 
[Sulfuras, Hand of Ragnaros, 0, null] => java.lang.NullPointerException: null 
[Sulfuras, Hand of Ragnaros, 0, 0] => Sulfuras, Hand of Ragnaros, 0, 0 
[Sulfuras, Hand of Ragnaros, 0, 1] => Sulfuras, Hand of Ragnaros, 0, 1 
[Sulfuras, Hand of Ragnaros, 0, 50] => Sulfuras, Hand of Ragnaros, 0, 50 
[Sulfuras, Hand of Ragnaros, 0, 49] => Sulfuras, Hand of Ragnaros, 0, 49 
[Sulfuras, Hand of Ragnaros, 0, 51] => Sulfuras, Hand of Ragnaros, 0, 51 
[Sulfuras, Hand of Ragnaros, -1, null] => java.lang.NullPointerException: null 
[Sulfuras, Hand of Ragnaros, -1, 0] => Sulfuras, Hand of Ragnaros, -1, 0 
[Sulfuras, Hand of Ragnaros, -1, 1] => Sulfuras, Hand of Ragnaros, -1, 1 
[Sulfuras, Hand of Ragnaros, -1, 50] => Sulfuras, Hand of Ragnaros, -1, 50 
[Sulfuras, Hand of Ragnaros, -1, 49] => Sulfuras, Hand of Ragnaros, -1, 49 
[Sulfuras, Hand of Ragnaros, -1, 51] => Sulfuras, Hand of Ragnaros, -1, 51 
[Sulfuras, Hand of Ragnaros, 10, null] => java.lang.NullPointerException: null 
[Sulfuras, Hand of Ragnaros, 10, 0] => Sulfuras, Hand of Ragnaros, 10, 0 
[Sulfuras, Hand of Ragnaros, 10, 1] => Sulfuras, Hand of Ragnaros, 10, 1 
[Sulfuras, Hand of Ragnaros, 10, 50] => Sulfuras, Hand of Ragnaros, 10, 50 
[Sulfuras, Hand of Ragnaros, 10, 49] => Sulfuras, Hand of Ragnaros, 10, 49 
[Sulfuras, Hand of Ragnaros, 10, 51] => Sulfuras, Hand of Ragnaros, 10, 51 
[Sulfuras, Hand of Ragnaros, 11, null] => java.lang.NullPointerException: null 
[Sulfuras, Hand of Ragnaros, 11, 0] => Sulfuras, Hand of Ragnaros, 11, 0 
[Sulfuras, Hand of Ragnaros, 11, 1] => Sulfuras, Hand of Ragnaros, 11, 1 
[Sulfuras, Hand of Ragnaros, 11, 50] => Sulfuras, Hand of Ragnaros, 11, 50 
[Sulfuras, Hand of Ragnaros, 11, 49] => Sulfuras, Hand of Ragnaros, 11, 49 
[Sulfuras, Hand of Ragnaros, 11, 51] => Sulfuras, Hand of Ragnaros, 11, 51 
[Sulfuras, Hand of Ragnaros, 12, null] => java.lang.NullPointerException: null 
[Sulfuras, Hand of Ragnaros, 12, 0] => Sulfuras, Hand of Ragnaros, 12, 0 
[Sulfuras, Hand of Ragnaros, 12, 1] => Sulfuras, Hand of Ragnaros, 12, 1 
[Sulfuras, Hand of Ragnaros, 12, 50] => Sulfuras, Hand of Ragnaros, 12, 50 
[Sulfuras, Hand of Ragnaros, 12, 49] => Sulfuras, Hand of Ragnaros, 12, 49 
[Sulfuras, Hand of Ragnaros, 12, 51] => Sulfuras, Hand of Ragnaros, 12, 51 
[Sulfuras, Hand of Ragnaros, 13, null] => java.lang.NullPointerException: null 
[Sulfuras, Hand of Ragnaros, 13, 0] => Sulfuras, Hand of Ragnaros, 13, 0 
[Sulfuras, Hand of Ragnaros, 13, 1] => Sulfuras, Hand of Ragnaros, 13, 1 
[Sulfuras, Hand of Ragnaros, 13, 50] => Sulfuras, Hand of Ragnaros, 13, 50 
[Sulfuras, Hand of Ragnaros, 13, 49] => Sulfuras, Hand of Ragnaros, 13, 49 
[Sulfuras, Hand of Ragnaros, 13, 51] => Sulfuras, Hand of Ragnaros, 13, 51 
[Sulfuras, Hand of Ragnaros, 14, null] => java.lang.NullPointerException: null 
[Sulfuras, Hand of Ragnaros, 14, 0] => Sulfuras, Hand of Ragnaros, 14, 0 
[Sulfuras, Hand of Ragnaros, 14, 1] => Sulfuras, Hand of Ragnaros, 14, 1 
[Sulfuras, Hand of Ragnaros, 14, 50] => Sulfuras, Hand of Ragnaros, 14, 50 
[Sulfuras, Hand of Ragnaros, 14, 49] => Sulfuras, Hand of Ragnaros, 14, 49 
[Sulfuras, Hand of Ragnaros, 14, 51] => Sulfuras, Hand of Ragnaros, 14, 51 
